LLK/scratch-vm
# Resolve project load error with undefined opcodes inside c/e blocks.

### Resolves

Project loading error for the first project mentioned in #1450.

### Proposed Changes

Fix issue where a c-block that contained only blocks with unknown opcodes, was causing an import error.

### Reason for Changes

#1254 skips unknown opcodes when importing blocks from sb2 projects. However, that PR missed the case where all of the statements inside a c-block are undefined. This PR fixes projects that were failing to load because of that issue.

### Test Coverage

Manually tested project 219549113 (mentioned in #1450). Tested some of the projects in the top 1000 projects list. Added fixture and integration test for example failing project.

cc/ @mzgoddard -- Just wanted to give you a heads up on this patch.


### How to test this
1. In Scratch 2.0, drag out a c-block (e.g. `forever`)
2. Plug in one or more blocks with unknown opcodes inside the c-block
(Examples are: `Lego WeDo 1.0` blocks, a custom procedure caller from a different project or sprite that came out of the backpack without its definition).
3. Don't plug in any blocks with known opcodes inside of the c-block (those already import correctly), but you may optionally plug these outside of the c-block if you wish.
4. Import the project into 3.0
The project *should* be able to load and there should be nothing inside the c-block (blocks with unknown opcodes get skipped when importing into 3.0). There *should* be warnings about skipped opcodes in the console.
